os/arch/arm/src/imxrt/imxrt_adc.c

```diff
@@ -860,4 +912,102 @@ struct adc_dev_s *imxrt_adc_initialize(int bus, int channel)
 
 	return &g_adcdev[channel];
 }
+#endif
+
+/************************************************************************************
+ * Public Data
+ ************************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: imxrt_gpio_configinput
+ ****************************************************************************/
+
+//static
+int imxrt_adc_configinput(uint32_t pinset)
+{
+	int bus = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
+	int channel = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
+	FAR const uint8_t *table;
+	iomux_pinset_t ioset;
+	uintptr_t regaddr;
+	unsigned int index;
+
+	DEBUGASSERT((unsigned int)bus < IMXRT_ADC_NBUS);
+
+	/* Configure pin as in input */
+
+//	imxrt_gpio_dirin(port, pin);		// ??
+
+	/* Configure pin as a GPIO */
+
+	table = g_adc_padmux[bus];
+	if (table == NULL) {
+		return -EINVAL;
+	}
+
+	index = (unsigned int)table[channel];
+	if (index >= IMXRT_PADMUX_NREGISTERS) {
+		return -EINVAL;
+	}
+
+	regaddr = imxrt_padmux_address(index);
+	putreg32(PADMUX_MUXMODE_ALT5, regaddr);
+
+//	imxrt_gpio_select(bus, channel);
+	/* Configure channel pad settings */
+
+	index = imxrt_padmux_map(index);
+	if (index >= IMXRT_PADCTL_NREGISTERS) {
+		return -EINVAL;
+	}
+
+	regaddr = imxrt_padctl_address(index);
+	ioset = (iomux_pinset_t)((pinset & GPIO_IOMUX_MASK) >> GPIO_IOMUX_SHIFT);
+	return imxrt_iomux_configure(regaddr, ioset);
+}
+
+FAR struct adc_dev_s * imxrt_adc_lowerhalf(uint32_t pinset)
+{
+	struct adc_dev_s *adcdev;
+	struct imxrt_dev_s *priv;
+	int bus = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
+	int channel = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
+
+	if (channel > IMXRT_ADC_NCHANEL) {
+		allvdbg("IMXRT ADC1 has maximum %d ADC channels.\n", IMXRT_ADC_NCHANEL);
+		return NULL;
+	}
+
+	adcdev = (struct adc_dev_s *)
+			kmm_malloc(sizeof(struct adc_dev_s));
```

Comment:
Please remove spaces or tabs between typecast and kmm_malloc.
like: adcdev = (struct adc_dev_s *)kmm_malloc(sizeof(struct adc_dev_s));

Reply:
thanks I'll fix